src/pystack/_pystack/mem.h

```diff
@@ -227,11 +232,11 @@ class CorefileRemoteMemoryManager : public AbstractRemoteMemoryManager
     std::shared_ptr<CoreFileAnalyzer> d_analyzer;
     std::vector<VirtualMap> d_vmaps;
     std::vector<SimpleVirtualMap> d_shared_libs;
+    char* corefile_data;
```

Comment:
@godlygeek I advised to start with the `CorefileRemoteMemoryManager` owning this pointer instead of encapsulating it into a RAII based thing, but this is something I wanted to check with you. We could move all the mmap logic to its own layer and handle the thing in a destructor so this class just holds a `unique_ptr` and that's it. Thoughts?

Reply:
Hm. I think it's fine having the destructor do it, but it doesn't seem like it would be hard to have RAII without adding a new mmap layer, unless I'm missing something. Can't we just declare it as:
```cpp
    std::unique_ptr<char, std::function<void(char*)>> corefile_data;
```
and initialize it as:
```cpp
    corefile_data = std::unique_ptr<char, std::function<void(char*)>>(
            reinterpret_cast<char*>(mmap(0, corefile_size, PROT_READ, MAP_PRIVATE, fd, 0)),
            [this](auto addr) {
                if (munmap(addr, corefile_size) == -1) {
                    LOG(ERROR) << "Failed to un-mmap a file " << d_analyzer-d_filename.c_str();
                }
            });
```
Though we'd need to sprinkle `.get()` everywhere where we use the pointer, which is a bit annoying, so 🤷 I guess on balance I'm fine with just calling `munmap` in the destructor.